GIAM-12095 Add GitHub workflows

andrzejdackiewicz opened the pull request.

Commit:
First version

Commit:
Initial github workflows implementation

grzegorz8 commented on .github/workflows/prepare-release.yml:
```diff
@@ -0,0 +1,57 @@
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
```
3.7 in every workflow. EMR has python 3.7.

andrzejdackiewicz replied:
Fixed

grzegorz8 commented on deployment-scripts/jobs-deployment/job_configuration.py:
```diff
@@ -40,7 +41,7 @@ def get_flink_properties(self) -> Dict[str, Any]:
 
     def get_flink_property(self, property_name: str) -> str:
-        return self.get_flink_properties().get(property_name)
+        return typing.cast(str, self.get_flink_properties().get(property_name))
```
why `typing.cast`?

andrzejdackiewicz replied:
![image](https://user-images.githubusercontent.com/88149093/221168218-9aa0dcd4-9bce-4705-a47a-9b0244b2f5d8.png)

grzegorz8 commented on deployment-scripts/jobs-deployment/tests/requirements.txt:
```diff
@@ -1,3 +1,6 @@
 mock>=2.0.0
-moto
\ No newline at end of file
+moto
```
provide version for moto

andrzejdackiewicz replied:
Added :)

grzegorz8 commented on deployment-scripts/jobs-deployment/tests/test_deploy_job.py:
```diff
@@ -12,9 +12,10 @@
-from deploy_job import EmrJobRunner, JinjaTemplateResolver
-from flink_clients import FlinkYarnRunner
-from job_configuration import JobConfiguration, JobConfigurationBuilder
+from deploy_job import EmrJobRunner, JinjaTemplateResolver  # noqa: 402
```
`noqa: 402`? what was wrong with the imports?

andrzejdackiewicz replied:
Flake8 will complain that these imports are not at the top of the file (because we use `sys.path.insert(...)` before.
If we remove the `sys.path.insert...` line then we will have trouble finding the imported classes. `noqa: 402` ignores that the import is not at the very top of file.

andrzejdackiewicz replied:
For example:
`deployment-scripts/jobs-deployment/tests/test_deploy_job.py:15:1: E402 module level import not at top of file`

grzegorz8 commented on setup.cfg:
```diff
@@ -0,0 +1,12 @@
+[tox:tox]
+envlist = py38
```
py37?

andrzejdackiewicz replied:
Changed to py37

grzegorz8 commented on setup.py:
```diff
@@ -0,0 +1,38 @@
+
+
+setup(
+    name="flink-sql-emr-runner-workflows",
```
`flink-sql-emr-runner-workflows` --> `flink-sql-runner`?
why workflows?

andrzejdackiewicz replied:
Derived from the name of the fork, changed to the flink-sql-runner as should be.

grzegorz8 commented on setup.py:
```diff
@@ -0,0 +1,38 @@
+    description="Framework for scheduling streaming SQL queries on Apache Hadoop YARN and on a standalone Flink cluster.",  # noqa: E501
+    long_description=README,
+    long_description_content_type="text/markdown",
+    python_requires=">=3.8",
```
3.7

andrzejdackiewicz replied:
Changed

Commit:
Resolving change requests

moss2k13 committed:
refactor workflows

moss2k13 commented:
I refactored github workflows as described below:
- opening PR towards `main` will triggerr tests and packaging
- merging PR towards `main` with proper release labels `release/{major/minor/patch}` will trigger release, changelog and upload to https://pypi.org
For https://pypi.org uploads github secret called `PYPI_API_TOKEN` needs to be added.

moss2k13 reviewed (approved):
github workflows verified successfully

moss2k13 committed:
increase artifact retention time

In case PR's review takes longer, artifact needs to be
available for release and publish process after the merge.
